modules/mod_search/support/search_query.erl

```diff
@@ -601,6 +547,41 @@ parse_query([Term|_], _Context, _Result) ->
 %% Helper functions
 %%
 
+%% @doc Parse hassubject and hasobject edges.
+-spec parse_edges(hassubject | hasobject, binary() | list(), #search_sql{}, z:context()) -> #search_sql{}.
+parse_edges(Term, Edges, Result, Context) when is_binary(Edges) ->
+    parse_edges(Term, maybe_split_list(Edges), Result, Context);
+parse_edges(Term, [[Id, Predicate]], Result, Context) ->
```

Comment:
So this was the bug solved here: with the new split behaviour introduced in #1866, values became nested in a list, so `[[Id, Predicate]]` instead of `[Id, Predicate]`.